### PR TITLE
Issue-204 Clientlibs Changes to remove FOUC

### LINKED ISSUE #204: Prevent FOUC with nav search on slow network
Currently there's a FOUC (flash-of-unstyled-content) for a few seconds when the search menu is opened on a slow 3g connection. Attaching a video capturing the same. This ticket is logged to prevent this. One possible solution could be along the lines of keeping the external-markup (including search markup which is fetched from Nedbank) hidden till all the styles have been loaded.

https://user-images.githubusercontent.com/1191451/232548933-f5abf31b-44d8-4615-9f08-c2fab0c432ea.mov

### PR DESCRIPTION
**Pull request by jindaliiita**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #204 

Test URLs:
- Before: https://main--nedbank--hlxsites.hlx.live/
- After: https://issue-204-temp--nedbank--hlxsites.hlx.live/

### CODE REVIEW

File: styles/styles.css [hunk 1]
```diff
@@ -85,6 +85,9 @@ body .modal {
 body .modal.fade {
   height: 0;
 }
+body .display-none {
```
Comment (vdua) [inline styles/styles.css:88]: nitpick: `body.hide` instead of `display-none`

File: blocks/header/nav-utils.js [hunk 2]
```diff
@@ -19,6 +27,20 @@ function appendStyles() {
...
+        cssLoaded();
+      };
+      style.onreadystatechange = () => {
+        // This will fire multiple times until css gets loaded
+        const { readyState } = this;
+        if (['loaded', 'complete'].indexOf(readyState) !== -1) {
+          style.onload();
+        }
+      };
+    }
```
Comment (sdmcraft) [inline blocks/header/nav-utils.js:43]: Is there a reason we need to do for all of these?
`onload`
`readyState=loaded` 
`readyState=complete` 

Would doing just for for one of these suffice?

File: blocks/header/nav-utils.js [hunk 3]
```diff
@@ -110,6 +131,7 @@ export async function loadNavTools() {
 
     const externalMarkup = document.createElement('div');
     externalMarkup.id = 'external-markup';
+    externalMarkup.classList.add('hide');
```
Comment (sdmcraft) [inline blocks/header/nav-utils.js:134]: I think we also need to keep the Search and Hamburger nav-tools hidden in the navbar as clicking on them does nothing till this markup remain hidden.

Comment (jindaliiita) [reply blocks/header/nav-utils.js:141]: Done

File: blocks/header/nav-utils.js (hunk 2 above)
Comment (jindaliiita) [reply blocks/header/nav-utils.js]: Done

**Review by sdmcraft — APPROVED** [review]

**Pull request merged**